ctapipe/io/zfits.py

```diff
@@ -4,115 +4,151 @@
-    container.meta.add_item('zfits__max_events', events)
-    container.meta.add_item('pixel_pos', dict())
-    container.add_item("r0", R0Container())
-    container.add_item("trig", CentralTriggerData())
-    container.add_item("count")
-
-    for run_id, event_id in eventstream:
-        container.dl0.run_id    = run_id
-        container.dl0.event_id  = event_id
-        #container.r0.event_num = zfits.get_event_number()
-
-        # We assume we are in the single-telescope case always:
-        container.dl0.tels_with_data     = [zfits.get_telescope_id(), ]
-        container.trig.tels_with_trigger = container.dl0.tels_with_data
-
-        time_s, time_ns = zfits.get_central_event_gps_time()
-        container.trig.gps_time = Time(time_s * u.s, time_ns * u.ns,
-                                       format='gps', scale='utc')
-
-        container.count = counter
-        t = np.arange(n_samples)
-        
-        container.dl0.tel = dict()  # clear the previous telescopes
-
-        # Depecrated loop, we keep it for clarity (similar structure than hessio and toymodel modules)
-        for tel_id in container.dl0.tels_with_data:
-            # fill pixel position dictionary, if not already done:
-            #TODO: tel_id here is a dummy parameter, we are dealing with single-telescope data!. TBR.
-            if tel_id not in container.inst.pixel_pos:
-                container.inst.pixel_pos[tel_id] = \
-                    zfits.get_pixel_position(tel_id) * u.m
-
-            nchans = zfits.get_num_channels(tel_id)
-            container.inst.num_channels[tel_id] = nchans
-            
-            for chan in range(nchans): 
-                samples = zfits.get_adc_sample(channel=chan, telescope_id=tel_id)
-                integrated = zfits.get_adc_sum(channel=chan, telescope_id=tel_id)
-                
-                container.dl0.tel[tel_id].pixel_samples[chan] = samples.keys()
-
-                mask = np.zeros(zfits.get_number_of_pixels(),dtype=bool)
-                mask[np.array(samples.keys())]=True
-                container.dl0.tel[tel_id].adc_samples[chan] = \
-                    ma.array(np.array(samples.values()),mask=mask)
-                
-                mask = np.zeros(zfits.get_number_of_pixels(),dtype=bool)
-                mask[np.array(integrated.keys())]=True
-                container.dl0.tel[tel_id].adc_integrated[chan] = \
-                    ma.array(np.array(integrated.values()),mask=mask)
-
-        yield container
-        counter +=1
-
-        if max_events is not None and counter > max_events:
-            return
-
+    telescope_ids = Field([], "list of IDs of telescopes used in the run")
+    num_pixels = Field(Map(int), "map of tel_id to number of pixels in camera")
+    num_channels = Field(Map(int), "map of tel_id to number of channels")
+    num_samples = Field(Map(int), "map of tel_id to number of samples")
+    geom = Field(Map(None), 'map of tel_if to CameraGeometry')
```

Comment:
I think `geom`, `cam`,  'telescope_ids`, and `optics` are not needed - they are all in the `SubarrayDescription`, which already works for SST-1M.    The only things that I see here that are *so far* SST-1M specific are the cluster and patch matrix definitions. All other info is already present elsewhere.

Reply:
By the way, what is the difference between cluster and patch? I assume one is trigger related and the other camera module. IN which case, we should use the correct CTA terminology from the glossary.

From the glossary:
- module: sub-component of camera containing several pixels
- cluster: a unit including several modules (min 2)
- patch: couldn't find definition in the glossary (probably should be one, or is there another name for trigger groups like "sector"?)

Reply:
Once I see how it works for SST1M, I think I'll just update the instrument module to have all this info - it shouldn't be in a Container (which are schemas for a row in an output table, and this data will not be written that way)

Reply:
Thank you for the feedback. This is really helpful. The more stuff, I can remove the better.

Reply:
> By the way, what is the difference between cluster and patch?
I had to look at that as well. Hopefully I get it right. It is basically described here on slide 6:
https://indico.cta-observatory.org/event/1362/contributions/14325/attachments/12437/14554/sst_1m_pipeline.pdf

This is all trigger related. 

The trigger sees a low resolution version of the camera, both in space and in dynamic range, but in real time of course. To make this low resolution image, 3 pixels are summed up and only the 8 least significant bits are transmitted. So instead of a 1296 X 16bit X 250 MHz = 5 Tbit/s stream, the Trigger "only" sees a ~860Gbit/s stream of data. On this data the trigger can take decisions on clusters of 7 patches or on 19 patches.